### ISSUE TITLE
Using Mean Effect Sizes metric as customized evaluation metric

### ISSUE THREAD

**Issue by vinaybenny** [dmlc/xgboost#1803]
I am trying to use xgboost as the predictive model for a propensity matching problem, which uses the Mean Effect Size of the covariates as the evaluation metric for the model. 

I'm trying to write a customised evaluation metric function in R that calculates the mean effect sizes of the covariates of the model, so that once I do a cross validation and hyperparameter search using xgb.cv function, I can pick the model with the lowest mean effect size.

Since the customised objective function is of the form "function(preds, dtrain)", and there is no way I can retrieve the original columns from the dtrain(which is a DMatrix object), I'm unable to calculate the mean effect sizes of the covariates in the model. Is there a workaround by which I can retrieve my original dataframe from dtrain within the customised eval metric?

Sample code below (Here, calcES is a function that calculates mean effect sizes from the modelling variables):

```
meaneffsize <- function(preds, dtrain) {

  es <- lapply(dtrain, calcES, t=getinfo(dtrain, "label"), w=preds/(1+preds), estimand="ATE");
  meaneffsize <- mean( abs( unlist( sapply(es,function(x){x$std.eff.sz}) ) ) )
  
  return(list(metric="meaneffsize", value= meaneffsize));
}

```
EDIT: Adding additional info. The calcES function uses the predictions made by the model to create sampling weights for the rows in the dataset. The aim is to get the treated and control groups to be balanced on the mean effect size for each covariate. This is different from a conventional classification problem of trying to predict the labels- here the labels indicate getting a treatment or intervention. This technique is used for making causal inferences.

**Comment by pommedeterresautee** [comment]
Old question, no answer. Closing
